### PR TITLE
Refine Risk Assessment Terms: Vulnerability and Impact Functions (Update Guide_Introduction.ipynb)

### PR DESCRIPTION
**Pull request by DahyannAraya**
The choice of terminology in risk assessment models such as CLIMADA is crucial, as it shapes the framework and perspective of the analysis. The distinction between "vulnerability function" and "impact function" is particularly significant. I made some changes along the lines of making this terminology more visible. I also made changes in the documentation where vulnerability function or damage function was still referenced in order to unify the communication of CLIMADA terminology. 

I considered adding a few lines to explain that the concepts of damage function, vulnerability function, and impact function, though interrelated, each serve a distinct purpose. Damage functions quantify specific damages, focusing on concrete impacts. Vulnerability functions, in contrast, articulate the susceptibility of different exposures to hazards. This susceptibility encompasses more than just the potential for damage; it includes a variety of preconditions that define an exposure's overall vulnerability. Meanwhile, impact functions assess the likelihood of a hazard's effects, both negative and positive, by parameterizing the degree to which an exposure will be impacted by a particular hazard.Something along those lines..... but maybe it's too much... we can discuss it sometime ......

Changes proposed in this PR:
- 
- 

This PR fixes #

### PR Author Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] Correct target branch selected (if unsure, select `develop`)
- [ ] Descriptive pull request title added
- [ ] Source branch up-to-date with target branch
- [ ] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [ ] [Tests][testing] updated
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]
- [ ] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/main/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

**Comment by chahank** [comment]
I think it is a great idea to clarify a bit the terminology. However, I think there are many more ways to interpret the words as those that you mentioned, so I would just be careful to not be to strong on the wording. The one thing we can be very clear about is what are `impact functions` in the context of CLIMADA. They are very flexible, and so can fulfill the roles of both `vulnerability functions` or `damage function`, but could also be `productivity functions` or `warning levels`. 

In any case, thanks for the updates!

File: doc/guide/Guide_Introduction.ipynb
Comment (zeliest) [inline doc/guide/Guide_Introduction.ipynb:15]: I would leave out this sentence here as it is a bit of a repetition of what comes below: "In CLIMADA, vulnerability is operationalized through the use of impact functions, which are essential in determining the extent to which exposure is affected by a given hazard."

File: doc/guide/Guide_Introduction.ipynb
Comment (zeliest) [inline doc/guide/Guide_Introduction.ipynb:44]: based on @chahank comment, I would modify the sentence "While the term 'vulnerability function' is broadly used in the modelers community, we adopt the term 'impact function' to explicitly include the option of opportunities (i.e. negative damages)" to something like "While the term 'vulnerability function' is broadly used in the modelers community, we adopt the broader term 'impact function'. Impact functions can be vulnerability functions or structural damage functions, but could also be productivity functions or warning levels. This definition also explicitly includes the option of opportunities (i.e. negative damages). "

File: doc/guide/Guide_Introduction.ipynb
Comment (zeliest) [inline doc/guide/Guide_Introduction.ipynb:46]: would make sense to have a new paragraph at "Using this approach..."

**Review by zeliest — CHANGES_REQUESTED** [review]
I think it is good idea to clarify the terminology here. I would however not go more into detail as it is a general introduction, maybe in the tutorial on impact functions?

**Comment by chahank** [comment]
@DahyannAraya : How is this PR going?

**Comment by DahyannAraya** [comment]
OK @chahank, it's ready. I have implemented the changes suggested by @zeliest. You can do the merge, or if not, I'll wait for your confirmation and do it myself... Thanks

**Comment by chahank** [comment]
Thanks @DahyannAraya ! But as you can see, all tests are failing. We need to fix this first. First thing to do is merge the `develop` branch into your branch as it has been a long time since this was done last. If tests are still failing, please try to fix them or seek help. 

@zeliest : can you please see if you are satisfied and approve the PR if yes (and if all tests are passing).

**Pull request merged**
